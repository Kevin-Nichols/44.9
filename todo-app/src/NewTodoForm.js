import React, { useState } from "react";
import { v4 as uuidv4 } from 'uuid';

function NewTodoForm({createTodo}) {
  const [todo, setTodo] = useState("");

  const handleChange = e => {
    setTodo(e.target.value);
  };

  const getInput = e => {
    e.preventDefault();
    createTodo({todo, id: uuidv4()});
    setTodo("");
  };

  return (
    <>
      <form onSubmit={getInput}>
        <label htmlFor="todo">Todo: </label>
        <input id="todo" name="todo" type="text" value={todo} onChange={handleChange} />
        <button>Add Todo</button>
      </form>
    </>
  );
}

export default NewTodoForm;