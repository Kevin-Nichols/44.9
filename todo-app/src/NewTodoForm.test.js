import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("is a smoke test for NewTodoForm", function() {
  render(<NewTodoForm />);
});

it("is a snapshot test for NewTodoForm", function() {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});