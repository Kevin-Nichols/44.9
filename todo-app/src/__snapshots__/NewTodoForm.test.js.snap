// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is a snapshot test for NewTodoForm 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo: 
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
